Cannot use Realm Kotlin classes in Java code

If referencing a Realm Kotlin class from a Java class like below

`Sample.kt`
```
class Sample: RealmObject {
    var id: String = "Sample"
}
```
And `Test.java`
```
public class Test {
    Sample  sample = new Sample();
}
```
Then compilation will fail with 
```
Test.java:4: error: cannot access Sample
    Sample  sample = new Sample();
    ^
  bad class file: Sample.class
    undeclared type variable: T
    Please remove or make sure it appears in the correct subdirectory of the classpath.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@
 
 ### Fixed
 * Allow defining properties with the field name as the persisted name. ([#1240](https://github.com/realm/realm-kotlin/issues/1240))
+* Fix compilation error when accessing Realm Kotlin model classed from Java code. ([#1256](https://github.com/realm/realm-kotlin/issues/1256))
 
 ### Compatibility
 * File format: Generates Realms with file format v23.

--- a/integration-tests/gradle-plugin-test/single-platform/src/main/java/io/realm/test/singleplatform/model/Pojo.java
+++ b/integration-tests/gradle-plugin-test/single-platform/src/main/java/io/realm/test/singleplatform/model/Pojo.java
@@ -1,0 +1,5 @@
+package io.realm.test.singleplatform.model;
+
+public class Pojo {
+    TestClass o = new TestClass();
+}

--- a/integration-tests/gradle-plugin-test/single-platform/src/main/java/io/realm/test/singleplatform/model/Pojo.java
+++ b/integration-tests/gradle-plugin-test/single-platform/src/main/java/io/realm/test/singleplatform/model/Pojo.java
@@ -1,5 +1,7 @@
 package io.realm.test.singleplatform.model;
 
+// This class is only added to verify that we can access the Realm Model classes from plain old
+// Java source. See issue for further details https://github.com/realm/realm-kotlin/issues/1256
 public class Pojo {
     TestClass o = new TestClass();
 }

--- a/packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt
+++ b/packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt
@@ -258,7 +258,7 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
             pluginContext,
             realmObjectCompanionInterface,
             REALM_OBJECT_COMPANION_CLASS_MEMBER,
-            pluginContext.irBuiltIns.kClassClass.defaultType
+            pluginContext.irBuiltIns.kClassClass.typeWith(clazz.defaultType)
         ) { startOffset, endOffset ->
             IrClassReferenceImpl(
                 startOffset = startOffset,

--- a/packages/plugin-compiler/src/test/resources/sample/expected/01_AFTER.ValidateIrBeforeLowering.ir
+++ b/packages/plugin-compiler/src/test/resources/sample/expected/01_AFTER.ValidateIrBeforeLowering.ir
@@ -5406,17 +5406,17 @@ MODULE_FRAGMENT name:<main>
             RETURN type=kotlin.Nothing from='public open fun io_realm_kotlin_newInstance (): kotlin.Any declared in sample.input.Sample.Companion'
               CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in sample.input.Sample' type=sample.input.Sample origin=null
         PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
-          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private
+          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.Sample> visibility:private
             EXPRESSION_BODY
               CLASS_REFERENCE 'CLASS CLASS name:Sample modality:OPEN visibility:public superTypes:[io.realm.kotlin.types.RealmObject; io.realm.kotlin.internal.RealmObjectInternal]' type=sample.input.Sample
-          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.Sample.Companion) returnType:kotlin.reflect.KClass<T of kotlin.reflect.KClass>
+          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.Sample.Companion) returnType:kotlin.reflect.KClass<sample.input.Sample>
             correspondingProperty: PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
             overridden:
               public abstract fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<out io.realm.kotlin.types.TypedRealmObject> declared in io.realm.kotlin.internal.RealmObjectCompanion
             $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:sample.input.Sample.Companion
             BLOCK_BODY
-              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<T of kotlin.reflect.KClass> declared in sample.input.Sample.Companion'
-                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private' type=kotlin.reflect.KClass<T of kotlin.reflect.KClass> origin=null
+              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<sample.input.Sample> declared in sample.input.Sample.Companion'
+                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.Sample> visibility:private' type=kotlin.reflect.KClass<sample.input.Sample> origin=null
                   receiver: GET_VAR '<this>: sample.input.Sample.Companion declared in sample.input.Sample.Companion.<get-io_realm_kotlin_class>' type=sample.input.Sample.Companion origin=null
         PROPERTY name:io_realm_kotlin_className visibility:public modality:FINAL [var]
           FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_className type:kotlin.String visibility:private
@@ -6102,17 +6102,17 @@ MODULE_FRAGMENT name:<main>
             RETURN type=kotlin.Nothing from='public open fun io_realm_kotlin_newInstance (): kotlin.Any declared in sample.input.Child.Companion'
               CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in sample.input.Child' type=sample.input.Child origin=null
         PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
-          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private
+          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.Child> visibility:private
             EXPRESSION_BODY
               CLASS_REFERENCE 'CLASS CLASS name:Child modality:OPEN visibility:public superTypes:[io.realm.kotlin.types.RealmObject; io.realm.kotlin.internal.RealmObjectInternal]' type=sample.input.Child
-          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.Child.Companion) returnType:kotlin.reflect.KClass<T of kotlin.reflect.KClass>
+          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.Child.Companion) returnType:kotlin.reflect.KClass<sample.input.Child>
             correspondingProperty: PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
             overridden:
               public abstract fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<out io.realm.kotlin.types.TypedRealmObject> declared in io.realm.kotlin.internal.RealmObjectCompanion
             $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:sample.input.Child.Companion
             BLOCK_BODY
-              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<T of kotlin.reflect.KClass> declared in sample.input.Child.Companion'
-                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private' type=kotlin.reflect.KClass<T of kotlin.reflect.KClass> origin=null
+              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<sample.input.Child> declared in sample.input.Child.Companion'
+                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.Child> visibility:private' type=kotlin.reflect.KClass<sample.input.Child> origin=null
                   receiver: GET_VAR '<this>: sample.input.Child.Companion declared in sample.input.Child.Companion.<get-io_realm_kotlin_class>' type=sample.input.Child.Companion origin=null
         PROPERTY name:io_realm_kotlin_className visibility:public modality:FINAL [var]
           FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_className type:kotlin.String visibility:private
@@ -6332,17 +6332,17 @@ MODULE_FRAGMENT name:<main>
             RETURN type=kotlin.Nothing from='public open fun io_realm_kotlin_newInstance (): kotlin.Any declared in sample.input.EmbeddedParent.Companion'
               CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in sample.input.EmbeddedParent' type=sample.input.EmbeddedParent origin=null
         PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
-          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private
+          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.EmbeddedParent> visibility:private
             EXPRESSION_BODY
               CLASS_REFERENCE 'CLASS CLASS name:EmbeddedParent modality:OPEN visibility:public superTypes:[io.realm.kotlin.types.RealmObject; io.realm.kotlin.internal.RealmObjectInternal]' type=sample.input.EmbeddedParent
-          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.EmbeddedParent.Companion) returnType:kotlin.reflect.KClass<T of kotlin.reflect.KClass>
+          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.EmbeddedParent.Companion) returnType:kotlin.reflect.KClass<sample.input.EmbeddedParent>
             correspondingProperty: PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
             overridden:
               public abstract fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<out io.realm.kotlin.types.TypedRealmObject> declared in io.realm.kotlin.internal.RealmObjectCompanion
             $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:sample.input.EmbeddedParent.Companion
             BLOCK_BODY
-              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<T of kotlin.reflect.KClass> declared in sample.input.EmbeddedParent.Companion'
-                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private' type=kotlin.reflect.KClass<T of kotlin.reflect.KClass> origin=null
+              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<sample.input.EmbeddedParent> declared in sample.input.EmbeddedParent.Companion'
+                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.EmbeddedParent> visibility:private' type=kotlin.reflect.KClass<sample.input.EmbeddedParent> origin=null
                   receiver: GET_VAR '<this>: sample.input.EmbeddedParent.Companion declared in sample.input.EmbeddedParent.Companion.<get-io_realm_kotlin_class>' type=sample.input.EmbeddedParent.Companion origin=null
         PROPERTY name:io_realm_kotlin_className visibility:public modality:FINAL [var]
           FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_className type:kotlin.String visibility:private
@@ -6550,17 +6550,17 @@ MODULE_FRAGMENT name:<main>
             RETURN type=kotlin.Nothing from='public open fun io_realm_kotlin_newInstance (): kotlin.Any declared in sample.input.EmbeddedChild.Companion'
               CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in sample.input.EmbeddedChild' type=sample.input.EmbeddedChild origin=null
         PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
-          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private
+          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.EmbeddedChild> visibility:private
             EXPRESSION_BODY
               CLASS_REFERENCE 'CLASS CLASS name:EmbeddedChild modality:OPEN visibility:public superTypes:[io.realm.kotlin.types.EmbeddedRealmObject; io.realm.kotlin.internal.RealmObjectInternal]' type=sample.input.EmbeddedChild
-          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.EmbeddedChild.Companion) returnType:kotlin.reflect.KClass<T of kotlin.reflect.KClass>
+          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:sample.input.EmbeddedChild.Companion) returnType:kotlin.reflect.KClass<sample.input.EmbeddedChild>
             correspondingProperty: PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
             overridden:
               public abstract fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<out io.realm.kotlin.types.TypedRealmObject> declared in io.realm.kotlin.internal.RealmObjectCompanion
             $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:sample.input.EmbeddedChild.Companion
             BLOCK_BODY
-              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<T of kotlin.reflect.KClass> declared in sample.input.EmbeddedChild.Companion'
-                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private' type=kotlin.reflect.KClass<T of kotlin.reflect.KClass> origin=null
+              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<sample.input.EmbeddedChild> declared in sample.input.EmbeddedChild.Companion'
+                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<sample.input.EmbeddedChild> visibility:private' type=kotlin.reflect.KClass<sample.input.EmbeddedChild> origin=null
                   receiver: GET_VAR '<this>: sample.input.EmbeddedChild.Companion declared in sample.input.EmbeddedChild.Companion.<get-io_realm_kotlin_class>' type=sample.input.EmbeddedChild.Companion origin=null
         PROPERTY name:io_realm_kotlin_className visibility:public modality:FINAL [var]
           FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_className type:kotlin.String visibility:private

--- a/packages/plugin-compiler/src/test/resources/schema/expected/01_AFTER.ValidateIrBeforeLowering.ir
+++ b/packages/plugin-compiler/src/test/resources/schema/expected/01_AFTER.ValidateIrBeforeLowering.ir
@@ -50,17 +50,17 @@ MODULE_FRAGMENT name:<main>
             RETURN type=kotlin.Nothing from='public open fun io_realm_kotlin_newInstance (): kotlin.Any declared in schema.input.A.Companion'
               CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in schema.input.A' type=schema.input.A origin=null
         PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
-          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private
+          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<schema.input.A> visibility:private
             EXPRESSION_BODY
               CLASS_REFERENCE 'CLASS CLASS name:A modality:OPEN visibility:public superTypes:[io.realm.kotlin.types.RealmObject; io.realm.kotlin.internal.RealmObjectInternal]' type=schema.input.A
-          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:schema.input.A.Companion) returnType:kotlin.reflect.KClass<T of kotlin.reflect.KClass>
+          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:schema.input.A.Companion) returnType:kotlin.reflect.KClass<schema.input.A>
             correspondingProperty: PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
             overridden:
               public abstract fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<out io.realm.kotlin.types.TypedRealmObject> declared in io.realm.kotlin.internal.RealmObjectCompanion
             $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:schema.input.A.Companion
             BLOCK_BODY
-              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<T of kotlin.reflect.KClass> declared in schema.input.A.Companion'
-                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private' type=kotlin.reflect.KClass<T of kotlin.reflect.KClass> origin=null
+              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<schema.input.A> declared in schema.input.A.Companion'
+                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<schema.input.A> visibility:private' type=kotlin.reflect.KClass<schema.input.A> origin=null
                   receiver: GET_VAR '<this>: schema.input.A.Companion declared in schema.input.A.Companion.<get-io_realm_kotlin_class>' type=schema.input.A.Companion origin=null
         PROPERTY name:io_realm_kotlin_className visibility:public modality:FINAL [var]
           FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_className type:kotlin.String visibility:private
@@ -202,17 +202,17 @@ MODULE_FRAGMENT name:<main>
             RETURN type=kotlin.Nothing from='public open fun io_realm_kotlin_newInstance (): kotlin.Any declared in schema.input.B.Companion'
               CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in schema.input.B' type=schema.input.B origin=null
         PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
-          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private
+          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<schema.input.B> visibility:private
             EXPRESSION_BODY
               CLASS_REFERENCE 'CLASS CLASS name:B modality:OPEN visibility:public superTypes:[io.realm.kotlin.types.RealmObject; io.realm.kotlin.internal.RealmObjectInternal]' type=schema.input.B
-          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:schema.input.B.Companion) returnType:kotlin.reflect.KClass<T of kotlin.reflect.KClass>
+          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:schema.input.B.Companion) returnType:kotlin.reflect.KClass<schema.input.B>
             correspondingProperty: PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
             overridden:
               public abstract fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<out io.realm.kotlin.types.TypedRealmObject> declared in io.realm.kotlin.internal.RealmObjectCompanion
             $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:schema.input.B.Companion
             BLOCK_BODY
-              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<T of kotlin.reflect.KClass> declared in schema.input.B.Companion'
-                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private' type=kotlin.reflect.KClass<T of kotlin.reflect.KClass> origin=null
+              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<schema.input.B> declared in schema.input.B.Companion'
+                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<schema.input.B> visibility:private' type=kotlin.reflect.KClass<schema.input.B> origin=null
                   receiver: GET_VAR '<this>: schema.input.B.Companion declared in schema.input.B.Companion.<get-io_realm_kotlin_class>' type=schema.input.B.Companion origin=null
         PROPERTY name:io_realm_kotlin_className visibility:public modality:FINAL [var]
           FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_className type:kotlin.String visibility:private
@@ -354,17 +354,17 @@ MODULE_FRAGMENT name:<main>
             RETURN type=kotlin.Nothing from='public open fun io_realm_kotlin_newInstance (): kotlin.Any declared in schema.input.C.Companion'
               CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in schema.input.C' type=schema.input.C origin=null
         PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
-          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private
+          FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<schema.input.C> visibility:private
             EXPRESSION_BODY
               CLASS_REFERENCE 'CLASS CLASS name:C modality:OPEN visibility:public superTypes:[io.realm.kotlin.types.RealmObject; io.realm.kotlin.internal.RealmObjectInternal]' type=schema.input.C
-          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:schema.input.C.Companion) returnType:kotlin.reflect.KClass<T of kotlin.reflect.KClass>
+          FUN DEFAULT_PROPERTY_ACCESSOR name:<get-io_realm_kotlin_class> visibility:public modality:FINAL <> ($this:schema.input.C.Companion) returnType:kotlin.reflect.KClass<schema.input.C>
             correspondingProperty: PROPERTY name:io_realm_kotlin_class visibility:public modality:FINAL [var]
             overridden:
               public abstract fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<out io.realm.kotlin.types.TypedRealmObject> declared in io.realm.kotlin.internal.RealmObjectCompanion
             $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:schema.input.C.Companion
             BLOCK_BODY
-              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<T of kotlin.reflect.KClass> declared in schema.input.C.Companion'
-                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<T of kotlin.reflect.KClass> visibility:private' type=kotlin.reflect.KClass<T of kotlin.reflect.KClass> origin=null
+              RETURN type=kotlin.Nothing from='public final fun <get-io_realm_kotlin_class> (): kotlin.reflect.KClass<schema.input.C> declared in schema.input.C.Companion'
+                GET_FIELD 'FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_class type:kotlin.reflect.KClass<schema.input.C> visibility:private' type=kotlin.reflect.KClass<schema.input.C> origin=null
                   receiver: GET_VAR '<this>: schema.input.C.Companion declared in schema.input.C.Companion.<get-io_realm_kotlin_class>' type=schema.input.C.Companion origin=null
         PROPERTY name:io_realm_kotlin_className visibility:public modality:FINAL [var]
           FIELD PROPERTY_BACKING_FIELD name:io_realm_kotlin_className type:kotlin.String visibility:private

